### ISSUE TITLE
Is there a reason OSX isn't supported for mpi?

### ISSUE THREAD

**Issue by sbromberger** [cpmech/gosl#20]
I just commented out all the build directives and removed `mpi_notlinux.go` and things appear to work on OSX 10.13.3 with openmpi 3.0.0 installed via homebrew.

Is there a reason I *shouldn't* do this? I've only tested a trivial program so far.

**Comment by sbromberger** [comment]
I just tested on a 4-core OSX machine (10.12.6) and `mpi/xrunmpitests.bash` passed all tests except for test 1 which requires 8 processors.

**Comment by cpmech** [comment]
It should work fine on macOS. I just haven't the time to fully test Gosl/MPI on that platform yet.

Could you please let me know which Homebrew package you had to install to get MPI working on macOS?

Many thanks!

**Comment by sbromberger** [comment]
Hi @cpmech - please see #22 - I made a PR with instructions in the comment. Basically, all I needed was `brew install openmpi`.

**Comment by cpmech** [comment]
Hi,

I just would like to announce that MPI (and HDF5!) seem to be working fine on my MacBook Air.

The macOS installation instructions have been updated slightly (with MPI and HDF5). Also, homebrew science has been deprecated. So, I've removed all homebrew packages and reinstalled again with the new version.

The MPI code can be tested with more processors than available by using `xrunmpitestsOS.bash` (OS means --oversubscribe). Not optimal of course.

Cheers.
Dorival
